### ISSUE TITLE
`-proxy` argument ignored for `-update-templates` option

### ISSUE THREAD

**Issue by hlein** [projectdiscovery/nuclei#3440]
### Nuclei version:
`2.8.9`

### Current Behavior:
Running in an isolated environment with `nuclei -proxy http://127.0.0.1:3128 -update-templates -v` fails with `[ERR] Could not update templates: Version string empty`

### Expected Behavior:
nuclei should use the named proxy, and successfully update the templates.

### Steps To Reproduce:
1. Arrange for an HTTP proxy such as `127.0.0.1:3128`
2. Run `nuclei -proxy http://127.0.0.1:3128 -update-templates -v`
3. Get an error:
```
                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.9

                projectdiscovery.io

[VER] Using http://127.0.0.1:3128 as proxy server
[WRN] Could not fetch latest releases: could not do request: Get "https://version-check.nuclei.sh/versions?v=2.8.9": context deadline exceeded
[INF] No new updates found for nuclei templates
```

### Anything else:
Setting the `https_proxy` environment variable (with or without providing a `-proxy ...` argument) succeeds:

```
$ https_proxy=http://127.0.0.1:3128 nuclei -update-templates -v

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.9

                projectdiscovery.io

[INF] No new updates found for nuclei templates
```

```
$ https_proxy=http://127.0.0.1:3128 nuclei -proxy http://127.0.0.1:3128 -update-templates -v

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.8.9

                projectdiscovery.io

[VER] Using http://127.0.0.1:3128 as proxy server
[INF] No new updates found for nuclei templates
```

This makes me think that while the `-proxy` parameter is parsed in all modes, its value is not consulted when fetching template updates.

I'm not sure what the impact of `-proxy-internal` is intended to be, but it does not change this behavior either.

**Comment by ehsandeep** [comment]
@hlein are you able to update the templates without using proxy?

**Comment by hlein** [comment]
> @hlein are you able to update the templates without using proxy?

I mean... no? This instance is isolated, like I said, no direct Internet access.

**Comment by ehsandeep** [comment]
@hlein, apologies for overlooking the initial part; in that case, this is expected as of now, as the update option doesn't go through the proxy option.

We will get it fixed by ensuring `-update` option respect the proxy input.

**Comment by hlein** [comment]
Gotcha. Cool, thanks!

**Comment by tarunKoyalwar** [comment]
this will be resolved with https://github.com/projectdiscovery/nuclei/issues/3428 . as it is default behaviour for all pd tools

**Comment by tarunKoyalwar** [comment]
@hlein , closing this since this is already available/reflect in latest release (v2.9.2) . 

cc: @ehsandeep
